plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3BlobContainer.java

```diff
@@ -176,67 +188,68 @@ public DeleteResult delete() throws IOException {
     }
 
     @Override
-    public void deleteBlobsIgnoringIfNotExists(List<String> blobNames) throws IOException {
+    public void deleteBlobsIgnoringIfNotExists(Iterator<String> blobNames) throws IOException {
         doDeleteBlobs(blobNames, true);
     }
 
-    private void doDeleteBlobs(List<String> blobNames, boolean relative) throws IOException {
-        if (blobNames.isEmpty()) {
-            return;
-        }
-        final Set<String> outstanding;
+    private void doDeleteBlobs(Iterator<String> blobNames, boolean relative) throws IOException {
+        final Iterator<String> outstanding;
```

Comment:
nit: can't we return early if the iterator is empty as we were doing before?

Reply:
sure done